account_invoice_split_invoice/wizard/account_invoice_split_wizard.py

```diff
@@ -0,0 +1,110 @@
+                inv_line_id = line[1]
+                line_values = line[2]
+                if line_values.get('split'):
+                    # The line will be attached in a new invoice
+                    inv_lines_to_update.append(inv_line_id)
+                    invoice_id = inv_lines_vals.get('invoice_id')
+                    if not invoice_id:
+                        invoice_id = self._copy_invoice(cr, uid, ids, context)
+                        inv_lines_vals['invoice_id'] = invoice_id
+        if inv_lines_to_update:
+            old_invoice = self.browse(cr, uid, ids[0], context).invoice_id
+            inv_line_obj.write(cr, uid, inv_lines_to_update, inv_lines_vals)
+            if old_invoice.state == 'open':
+                # Cancel and validate again invoices
+                self.pool.get('account.invoice').action_cancel(
+                    cr, uid, [old_invoice.id], context=context)
+                self.pool.get('account.invoice').action_cancel_draft(
+                    cr, uid, [old_invoice.id])
+                wf_service = netsvc.LocalService('workflow')
+                wf_service.trg_validate(
+                    uid, 'account.invoice', old_invoice.id, 'invoice_open', cr)
+                wf_service.trg_validate(
+                    uid, 'account.invoice', invoice_id, 'invoice_open', cr)
+        return True
+
+    def _copy_invoice(self, cr, uid, ids, context=None):
+        # Create new invoice
+        invoice_obj = self.pool.get('account.invoice')
+        old_invoice = self.browse(cr, uid, ids[0], context).invoice_id
+        invoice_obj.copy(cr, uid, old_invoice.id, {
+            'date_invoice': old_invoice.date_invoice})
+        found_ids = invoice_obj.search(cr, uid, [
+            ('partner_id', '=', old_invoice.partner_id.id),
+            ('date_invoice', '=', old_invoice.date_invoice),
+            ('state', '=', 'draft')], context=context)
+        if found_ids and len(found_ids) == 1:
+            # Empty the new invoice
+            inv_line_obj = self.pool.get('account.invoice.line')
+            line_ids = inv_line_obj.search(cr, uid, [
+                ('invoice_id', '=', found_ids[0])], context=context)
+            inv_line_obj.unlink(cr, uid, line_ids, context)
+            return found_ids[0]
+        else:
+            raise orm.except_orm(
+                _('New invoice not found'),
+                _('The new invoice was not found so the process was '
+                  'cancelled.'))
+
+    _columns = {
+        'invoice_id': fields.many2one('account.invoice', 'Invoice'),
+        'invoice_line_ids': fields.function(
+            _get_invoice_line_ids, fnct_inv=_write_lines, type='one2many',
+            obj='account.invoice.line', method=True,
+            string=_('Invoice lines')),
+    }
+
+    _defaults = {
+        'invoice_id': _get_invoice_id,
+        'invoice_line_ids': _get_default_ids,
+    }
+
+    def split_invoice(self, cr, uid, ids, context=None):

```

Comment:
When the wizard is closed, it calls the _write_lines (when lines are changed). I don't know if I can have a save button that can close the wizard without calling any function but triggering the write. Maybe you have an idea or I will look if I find something.